reporters/drop-ship.es

```diff
@@ -81,16 +87,25 @@ export default class DropShipReporter extends BaseReporter {
       drop.baseExp = body.api_get_base_exp
       drop.shipId = (body.api_get_ship || {}).api_ship_id || -1
       drop.itemId = (body.api_get_useitem || {}).api_useitem_id || -1
+      drop.shipCounts = drop.shipId !== -1 ? countOwnedShipForms(this.ownedShipIds, drop.shipId) : []
       drop.teitokuLv = _teitokuLv
-      this.report('/api/report/v2/drop_ship', drop)
+      drop.teitokuId = teitokuId
+      // Report enemy pattern and drops
+      this.report('/api/report/v2/drop_ship', drop).then(() => {
+        this.drop = null
+      })
       // Report pass event
       if (body.api_get_eventitem != null) {
         this.report('/api/report/v2/pass_event', {
-          teitokuId: _teitokuId,
+          teitokuId,
           teitokuLv: _teitokuLv,
-          teitoku: _nickName,
           mapId: drop.mapId,
           mapLv: drop.mapLv,
+          rewards: body.api_get_eventitem.map(e => ({
```

Comment:
will `api_get_eventitem` be array if only 1 item is rewarded? we may need something like this: https://github.com/poooi/poi/blob/5825090930ff7d39372be3f3968a705d8a83a078/views/redux/info/equips.es#L12

Reply:
I think it is always an array in `mian.js`
```js
          var t = o.ObjUtil.getObjectArray(this._o, "api_get_eventitem");
          if (null == t) return [];
          for (var e = [], i = 0, n = t; i < n.length; i++) {
            var r = n[i];
            e.push(new u.MapBonusModel(r))
          }
      function getObjectArray(e, i, n) {
        void 0 === n && (n = null);
        var o = t._getProp(e, i, n);
        return o || n
      }
      function _getProp(t, e, i) {
        if (null != t && t.hasOwnProperty(e)) {
          var n = t[e];
          if (null != n && void 0 != n) return n
        }
        return i
      }
```

Reply:
Though just in case `isArray` can be used to prevent `map` throwing in case they change it (will throw last so will only disable pass_event reports).

Reply:
you're right, we could assume it is array

I'm asking because because they have done similar thing in API before. and I don't want to have any pass event report blocked because it is difficult to find out